tests/rbd/test_rbd_encryption.py

```diff
@@ -0,0 +1,163 @@
+                "nounmap": True,
+            },
+            "runtime": 30,
+            "device_names": [out],
+        }
+        kw["config"].update(io_config)
+        kw.update({"rbd_obj": rbd, "do_not_create_image": True})
+        krbd_io_handler(**kw)
+
+        log.info(f"Creating snap and clone for {image_spec}")
+        if rbd.snap_create(pool, image, snap):
+            log.error(f"Snapshot with name {snap} creation failed for {image_spec}")
+
+        if rbd.protect_snapshot(snap_spec):
+            log.error(f"Snapshot protect failed for {image_spec}")
+
+        krbd_io_handler(**kw)
+
+        if rbd.create_clone(snap_spec, pool, clone):
+            log.error(f"Clone creation failed for {clone_spec}")
+
+        log.info(
+            f"Applying encryption of type {encryption_type} on the clone {clone_spec}"
+        )
+        clone_passphrase = "clone_luks1_passphrase.bin"
+        rbd.create_passphrase_file(clone_passphrase)
+        if rbd.encrypt(clone_spec, encryption_type, clone_passphrase):
+            log.error(f"Apply RBD encryption {encryption_type} failed on {image_spec}")
+
+        encryption_config.append({encryption_type: clone_passphrase})
+        out, err = rbd.device_map(clone_spec, "nbd", reversed(encryption_config))
+        log.info(f"Out, Err : {out}: {err}")
+        if err:
+            log.error(
+                f"RBD device map failed for {image_spec} and encryption_config: {encryption_config}"
+            )
+
+        kw["config"]["device_names"].append(out)
+        kw["config"]["image_spec"].append(clone_spec)
+        krbd_io_handler(**kw)
+
+        flatten_config = {"encryption_config": reversed(encryption_config)}
+        if rbd.flatten_clone(pool, clone, **flatten_config):
+            log.error(f"Flatten clone with encryption failed for {clone_spec}")
+
+        log.info(
+            f"Verifying data integrity between original image: {image_spec} and clone: {clone_spec}"
+        )
+        check_data(rbd, image_spec, clone_spec)
+
+    except RbdBaseException as error:
+        print(error.message)
+        return 1
+
+
+def cleanup(rbd_obj, **kw):
+    """Cleanup the replicated and ecpools created for the test module
+
+    Args:
+        rbd_obj (dict): dictionary containing rbd ecpool and replicated pool object
+    """
+    log.info("Cleaning up pools")
+    obj = rbd_obj.get("rbd_ecpool", rbd_obj["rbd_reppool"])
+    obj.clean_up(
+        pools=[
+            kw["config"].get("rep_pool_config", {})["pool"],
+            kw["config"].get("ec_pool_config", {})["pool"],
+        ]
+    )
+
+
+def run(**kw):
+    """Verify the luks encryption functionality for image and clone
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common, fio and rbd-nbd packages,
+    conf and keyring files
+
+    Test case covered -
+    1) CEPH-83575263 - Encrypt & decrypt file using same keys and different keys
+    Test Case Flow
+    1. Create a pool and an Image
```

Comment:
I think "create snapshot , protect snapshot and create clone"  step is missing after step 1, 
correct me if i am not wrong as Step 2 says "apply encryption LUKS1 format to the clone image".

Reply:
Done.